web/controllers/password_reset_controller.ex

```diff
@@ -11,14 +11,14 @@ defmodule CodeCorps.PasswordResetController do
   and return email. 422 if pwd do not match or auth token does not exist
   """
   def reset_password(conn, %{"token" => token, "password" => password, "password_confirmation" => password_confirmation}) do
-    user = conn.assigns.current_user
-    with %AuthToken{value: auth_token} <- Repo.get_by(CodeCorps.AuthToken, %{ value: token, user_id: user.id }),
+    with %AuthToken{value: auth_token, user_id: user_id} <- Repo.get_by(CodeCorps.AuthToken, %{ value: token }),
       {:ok, _} <- Phoenix.Token.verify(CodeCorps.Endpoint, "user", auth_token, max_age: 1209600) do
-        with %Changeset{valid?: true} <- User.reset_password_changeset(user, 
-                                                                       %{password: password, password_confirmation: password_confirmation}) do
+        user = Repo.get(User, user_id)
+        with %Changeset{valid?: true} = changeset <- User.reset_password_changeset(user, 
+                                                                       %{password: password, password_confirmation: password_confirmation}),
+             {:ok, user} <- Repo.update(changeset) do
           conn
-          |> put_status(:created)
-          |> render("show.json", email: user.email)
+          |> TokenController.create(%{"username" => user.email, "password" => password})
```

Comment:
@snewcomer My proposal was, extract the logic from the token controller into a module. Then both the `TokenController` and the `PasswordResetController` call the same function in a third module.

That would eliminate the problem with `TokenController` using it's own view.

Looking at it now, thought, the part you need from `TokenController.create` is

```Elixir
{:ok, token, _claims} = user |> Guardian.encode_and_sign(:token)

conn
|> Plug.Conn.assign(:current_user, user)
|> put_status(:created)
|> render("show.json", token: token, user_id: user.id, email: user.email)
```

I would just put that smaller part of the code here, instead of calling `TokenController.create`